boa/src/syntax/parser/expression/mod.rs

```diff
@@ -489,9 +489,9 @@ where
     fn parse(self, cursor: &mut Cursor<R>, interner: &mut Interner) -> ParseResult {
         let _timer = BoaProfiler::global().start_event("Relation Expression", "Parsing");
 
-        if None::<InputElement>.is_some() {
-            cursor.set_goal(None::<InputElement>.unwrap());
-        }
+        // if None::<InputElement>.is_some() {
+        //     cursor.set_goal(None::<InputElement>.unwrap());
+        // }
```

Comment:
What was the point of this?

Reply:
Oh, leftover code, I was just surprised about the `None.is_some()` check, I will remove it

Reply:
But shouldn't it panic unconditionally? `None` is obviously not `Some`

Reply:
It panics always, but it checks before if it's some, so it never enters the branch.